libs/generic-view/ui/src/lib/buttons/button-base.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import React, { FunctionComponent, HTMLAttributes } from "react"
+import styled from "styled-components"
+import {
+  closeAllModals,
+  closeDomainModals,
+  closeModal,
+  openModal,
+  replaceModal,
+} from "generic-view/store"
+import { useDispatch } from "react-redux"
+import { useHistory } from "react-router"
+import { ButtonAction, useScreenTitle } from "generic-view/utils"
+
+interface Props extends HTMLAttributes<HTMLButtonElement> {
+  action: ButtonAction
+  viewKey?: string
+}
+
+export const ButtonBase: FunctionComponent<Props> = ({ action, ...props }) => {
```

Comment:
🟢 
The "Base Button" in the UI library serves as a raw button component, providing minimal complexity and independence for easy integration across various parts of the project.

Dependency on Context:
The component heavily relies on `useDispatch` and `useHistory`, potentially posing challenges for testing and maintenance. Suggestion: Consider separating these dependencies for increased flexibility.

Lack of Action Translation:
Action texts, like `open-modal` or `close-modal`, are directly embedded in the code, making interface localization challenging. A recommended consideration is to create an enum or utilize external resources to store these texts. This approach would significantly ease translation management for diverse languages.

Missing `onClick` Handling:
The absence of direct support for onClick limits the component's flexibility in user interactions. Suggestion: Adding `onClick` support would enable more diverse usage in different contexts.